### ISSUE TITLE
[v4.0.5.5981] DryIoc.ContainerException: code: Error.ContainerIsDisposed

### ISSUE THREAD

**Issue by gingerbeardman** [Radarr/Radarr#7141]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

- radarr quits after some days

### Expected Behavior

- should stay up forever

### Steps To Reproduce

1. install [v4.0.5.5981]
2. wait for some days

### Environment

```markdown
- OS: Synology 6.2.3 update 3
- Radarr: 4.0.5.5981 develop
```

### What branch are you running?

Develop

### Trace Logs?

```
2022-03-15 21:00:28.2|Error|TaskExtensions|Task Error

[v4.0.5.5981] DryIoc.ContainerException: code: Error.ContainerIsDisposed;
message: Container is disposed and should not be used: "container without scope has been DISPOSED!
 You may include Dispose stack-trace into the message via:
container.With(rules => rules.WithCaptureContainerDisposeStackTrace())"
   at DryIoc.Throw.It(Int32 error, Object arg0, Object arg1, Object arg2, Object arg3) in /_/src/DryIoc/Container.cs:line 13773
   at DryIoc.Container.ThrowIfContainerDisposed() in /_/src/DryIoc/Container.cs:line 684
   at DryIoc.Container.ResolveAndCache(Int32 serviceTypeHash, Type serviceType, IfUnresolved ifUnresolved) in /_/src/DryIoc/Container.cs:line 422
   at DryIoc.Container.DryIoc.IResolver.Resolve(Type serviceType, IfUnresolved ifUnresolved) in /_/src/DryIoc/Container.cs:line 360
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at NzbDrone.Core.ThingiProvider.ProviderFactory`2.GetInstance(TProviderDefinition definition) in D:\a\1\s\src\NzbDrone.Core\ThingiProvider\ProviderFactory.cs:line 132
   at System.Linq.Enumerable.SelectListIterator`2.ToList()
   at NzbDrone.Core.ThingiProvider.ProviderFactory`2.GetAvailableProviders() in D:\a\1\s\src\NzbDrone.Core\ThingiProvider\ProviderFactory.cs:line 92
   at NzbDrone.Core.Download.DownloadClientProvider.GetDownloadClients() in D:\a\1\s\src\NzbDrone.Core\Download\DownloadClientProvider.cs:line 91
   at NzbDrone.Core.HealthCheck.Checks.DownloadClientCheck.Check() in D:\a\1\s\src\NzbDrone.Core\HealthCheck\Checks\DownloadClientCheck.cs:line 28
   at NzbDrone.Core.HealthCheck.HealthCheckService.<>c.<PerformHealthCheck>b__12_0(IProvideHealthCheck c) in D:\a\1\s\src\NzbDrone.Core\HealthCheck\HealthCheckService.cs:line 75
   at System.Linq.Enumerable.SelectArrayIterator`2.ToList()
   at NzbDrone.Core.HealthCheck.HealthCheckService.PerformHealthCheck(IProvideHealthCheck[] healthChecks) in D:\a\1\s\src\NzbDrone.Core\HealthCheck\HealthCheckService.cs:line 75
   at NzbDrone.Core.HealthCheck.HealthCheckService.HandleAsync(ApplicationStartedEvent message) in D:\a\1\s\src\NzbDrone.Core\HealthCheck\HealthCheckService.cs:line 114
   at NzbDrone.Core.Messaging.Events.EventAggregator.<>c__DisplayClass6_2`1.<PublishEvent>b__2() in D:\a\1\s\src\NzbDrone.Core\Messaging\Events\EventAggregator.cs:line 121
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)


2022-03-15 21:00:29.5|Info|ConsoleApp|Exiting main.
```

[AB#2811](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/2811)

**Comment by gingerbeardman** [comment]
Seems to be this: https://github.com/dadhi/DryIoc/issues/435

**Comment by bakerboy448** [comment]
Just the error is not helpful. What was unclear about trace logs being required and the linked wiki article and how can they be improved or clarified?

This doesn't appear to be a Radar issue; this seems to be a Synology NAS issue. The SynoCommunity who create, update, maintain, and support the frequently out of date and rarely updated NAS Packages would be who would know Synology best.

**Comment by gingerbeardman** [comment]
Trace logs say that they should only be enabled temporarily, and I cannot reproduce this error on demand and it happens randomly after many days. That said, I'm happy to switch it on. But I only found out about the trace log requirement when arriving here to report the issue. So it's sort of chicken and egg. The bug has happened without trace logs.

You can see from my bug report that I'm running the very latest dev version of Radarr. Or do I misunderstand something about your claim that somehow I am running out of date code?

Finally, I sense some hostility. I'm trying to help to improve things for everybody.

**Comment by austinwbest** [comment]
> Trace logs say that they should only be enabled temporarily, and I cannot reproduce this error on demand and it happens randomly after many days. That said, I'm happy to switch it on. But I only found out about the trace log requirement when arriving here to report the issue. So it's sort of chicken and egg. The bug has happened without trace logs.
> 
> You can see from my bug report that I'm running the very latest dev version of Radarr. Or do I misunderstand something about your claim that somehow I am running out of date code?
> 
> Finally, I sense some hostility. I'm trying to help to improve things for everybody.

You're fine leaving trace on bud. Run it with it on until you get it reproduced.

**Comment by gingerbeardman** [comment]
I've switched it on and will report back here next time it goes down. Cheers

**Comment by Qstick** [comment]
> Seems to be this: [dadhi/DryIoc#435](https://github.com/dadhi/DryIoc/issues/435)

Thanks for finding this, I’ll bump our library version today to pull in their fix in 4.8.7.

**Comment by gingerbeardman** [comment]
My pleasure, thank you.
